CONTRIBUTING.md

```diff
@@ -90,6 +91,10 @@ and additionally the following rules:
   Keep in mind that the text in the github pull request description is only
   visible on github, not in the git log which can be considered permanent
   information storage.
+* Add comments to the source code if the code is not self-explanatory:
+  Comments in the source code should describe the choices made, to answer the
+  question "why is the code like this". The git commit message should describe
+  "why did we change it".
```

Comment:
How about also adding this?
```suggestion
  "why did we change it". If appropriate, consider adding some historical context to source code comments as well if this helps to understand the code (git commit messages can become quite hard to find when code snippets get shuffled around).
```

Reply:
so you want people to write more comments in source code because you don't know `git log -p --follow $filename`? ;)

Reply:
Well, I haven't known of this feature (thanks btw!) and I assume that neither will many others. But I get your point, so let's just omit it.